### PR TITLE
Variables: Page tweaks and updates

### PR DESCRIPTION
**Pull request by mottaquikarim**
- added one code block to explicitly display to user good/bad variable names
- added comments under the console.log statements that reflect output
- added optional challenge on the bottom of the page to get students to engage with the increment/decrement operators

### CODE REVIEW

File: lecture-variables.html [hunk 1]
```diff
@@ -63,23 +63,36 @@
...
         <li>The name of a variable is very important for you and others reading your code.</li>
-        <li>The name of a variable can contain letters, numbers, underscores, and dollar signs. However it <i>cannot start with a number</i>.
+        <li>The name of a variable can contain letters, numbers, underscores, and dollar signs. However it <i>cannot start with a number</i>. Also, <b>spaces</b> and <b>dashes</b> are <b>not</b> allowed in a variable name &mdash; having these characters in your variable name will throw an error.
       </ul>
     </p>
 
+    <pre><code>// BAD variable names
+var 3.14isPi; // can't start with number, can't have periods
+var hello everyone!; // space and '!' not allowed
+var this-is-also-not-kosher; // has dashes, too long
```
Comment (bensussman) [inline lecture-variables.html:90]: "too long" is an opinion. An important one, but an opinion none-the-less. We should make that clear

File: lecture-variables.html [hunk 2]
```diff
@@ -119,11 +133,11 @@
 someNumber /= x
 </code></pre>
 
-
+(Optional) <b>Challenge</b>: what is the value of <code>someNumber</code> at the end of the above code snippet? Walk through each line of code and try to compute the value of <code>someNumber</code> yourself. Then, paste the snippet above into codepen to check your answer.
```
Comment (bensussman) [inline lecture-variables.html:136]: This should be wrapped in a `<p>` tag. I think i'll merge this and make the change on master in the interest of speed

Comment (bensussman) [inline lecture-variables.html:136]: A bigger problem with this is that `x` is undefined. So this code won't run. Do you want me to give `x` a value so this code is actually valid?

Comment (mottaquikarim) [inline lecture-variables.html:136]: 5! let x = 5. Do you want me to submit a new PR for this? Since you've merged already.

Comment (bensussman) [inline lecture-variables.html:136]: naw i got it. pushing that change now.

File: lecture-variables.html [hunk 3]
```diff
@@ -63,23 +63,36 @@
...
+// Student's favorite language was French
 
 // .... after this lecture ....
 studentsFavLanguage = bensFavLanguage
 console.log("Student's favorite language was " + studentsFavLanguage)
-// Now it's JavaScript!</code></pre>
+// Student's favorite language was JavaScript</code></pre>
+
+    <p>
+        Notice how in the code snippet above our second <code>console.log</code> statement is identical to the first <code>console.log</code> statement but the output is drastically different since the value of the var <code>studentsFavLanguage</code> has changed.
```
Comment (bensussman) [inline lecture-variables.html:74]: Oh @mottaquikarim FYI i am using tabs-not-spaces and "two spaces" as my indent. Make sure to update your editor accordingly so we have consistent indentation

**Pull request merged**